components/BridgeActionsPair.tsx

```diff
@@ -0,0 +1,172 @@
+import { useBridge } from "@/providers/BridgeProvider";
+import { IntrinsicElements } from "@/types";
+import { css } from "@mui/material";
+import { useCallback, useEffect, useState, VFC } from "react";
+import SelectInput from "@/components/shared/SelectInput";
+import SwitchButton from "@/components/shared/SwitchButton";
+import { MenuItem, Theme } from "@mui/material";
+import { useCENNZWallet } from "@/providers/CENNZWalletProvider";
+
+interface BridgeActionsPairProps {}
+
+const BridgeActionsPair: VFC<
+	IntrinsicElements["div"] & BridgeActionsPairProps
+> = (props) => {
+	const selectOptions = [
+		{
+			value: "Ethereum",
+			label: "ETHEREUM",
+		},
+		{
+			value: "CENNZnet",
+			label: "CENNZnet",
+		},
+	];
+
+	const { bridgeAction, setBridgeAction, updateMetaMaskBalances } = useBridge();
+	const { updateBalances: updateCENNZBalances } = useCENNZWallet();
+	const [fromOption, setFromOption] = useState<string>("Ethereum");
+	const [toOption, setToOption] = useState<string>("CENNZnet");
+
+	const setOptionsPair = useCallback((fromOption: string, toOption: string) => {
+		if (fromOption) {
+			setFromOption(fromOption);
+			if (fromOption === "Ethereum") return setToOption("CENNZnet");
+			if (fromOption === "CENNZnet") return setToOption("Ethereum");
+		}
+
+		if (toOption) {
+			setToOption(toOption);
+			if (toOption === "Ethereum") return setFromOption("CENNZnet");
+			if (toOption === "CENNZnet") return setFromOption("Ethereum");
+		}
+	}, []);
+
+	const onFromActionChange = useCallback(
+		(event) => {
+			setOptionsPair(event.target.value, null);
+		},
+		[setOptionsPair]
+	);
+
+	const onToActionChange = useCallback(
+		(event) => {
+			setOptionsPair(null, event.target.value);
+		},
+		[setOptionsPair]
+	);
+
+	const onSwitchClick = useCallback(() => {
+		if (fromOption === "Ethereum") return setOptionsPair("CENNZnet", null);
+		if (fromOption === "CENNZnet") return setOptionsPair("Ethereum", null);
+	}, [fromOption, setOptionsPair]);
+
+	useEffect(() => {
+		if (fromOption === "Ethereum") return setBridgeAction("Deposit");
+		if (fromOption === "CENNZnet") return setBridgeAction("Withdraw");
+	}, [fromOption, setBridgeAction]);
+
+	useEffect(() => {
+		if (!fromOption) return;
+		if (fromOption === "Ethereum") return updateMetaMaskBalances?.();
+		if (fromOption === "CENNZnet") return updateCENNZBalances?.();
+	}, [fromOption, updateMetaMaskBalances, updateCENNZBalances]);
```

Comment:
what's with the ?.() syntax? is there a situation where these functions don't exist yet?

Reply:
it's part of optional-chaining, basically it's a short hand for
`if (updateMetaMaskBalances) updateMetaMaskBalances();`